frontend/client/src/components/ResetPasswordModal.js

```diff
@@ -0,0 +1,161 @@
+import React, { createRef } from 'react'
+import Modal from '../components/Modal'
+import PendingOperationButton from '../components/PendingOperationButton'
+import { auth } from '../store/firebase'
+import { withStore } from '../store'
+import Logging from '../util/logging'
+import firebase from 'firebase'
+
+class ResetPasswordModalBase extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      currentPassword: '',
+      password: '',
+      confirmPassword: '',
+      passwordsMatch: true,
+      passwordIsValid: false,
+      successful: false,
+      message: '',
+      formHasBeenEdited: false,
```

Comment:
I would rename this to `newPasswordHasBeenEdited` and add another state variable `currentPasswordHasBeenEdited` to indicate whether the user has changed the `current-password` field, so we can also avoid displaying an error message before the field is changed.